File: sorting/SelectionSort/cpp/SelectionSort.cpp
#include "SelectionSort.hpp"

#include <utility>

using namespace std;

void selectionSort(vector<int>& values) {
    const int n = static_cast<int>(values.size());

    for (int i = 0; i < n; ++i) {
        int minIndex = i;
        for (int j = i + 1; j < n; ++j) {
            if (values[j] < values[minIndex]) {
                minIndex = j;
            }
        }

        if (minIndex != i) {
            swap(values[i], values[minIndex]);
        }
    }
}

File: sorting/SelectionSort/rust/src/selection_sort.rs
pub fn selection_sort(values: &mut [i32]) {
    let n = values.len();

    for i in 0..n {
        let mut min_index = i;
        for j in (i + 1)..n {
            if values[j] < values[min_index] {
                min_index = j;
            }
        }

        if min_index != i {
            values.swap(i, min_index);
        }
    }
}
